### DISCUSSION TITLE
Cors blocking requests to openAPI from react frontend

### DISCUSSION THREAD

**Discussion by roopeshshettyb** [Netflix/conductor#3008] [Q&A]
Hi,

I am trying to create UI to display the workflow on my app using react as frontend. I am able to make calls with no problem from my backend. But when I try to call the openapi from my frontend directly I get an Axios Error/CORS Error

`Access to XMLHttpRequest at 'http://localhost:8080/api/workflow/51045a54-aa7f-4378-924b-710ae3ffbd03' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

I know a workaround would be calling my backend API and then calling the openapi but I feel it would be better if there's any way to fix this so my frontend can directly call the openapi

I read online that this can be fixed by adding the cors config on the conductor server, I tried snooping but couldn't figure how to. 
Could someone help me with this please?

**Comment by v1r3n** [comment]
CORS headers are added in the server - the alternative solution here is to reverse proxy to conductor server using nginx or react so Conductor server APIs are available on the same host:port as UI.

**Reply by roopeshshettyb** [reply]
Oh, okay. I did see some answers on how to use a proxy but is there no way I can change some config in the conductor files to add the headers?
